core/v2.go

```diff
@@ -2293,22 +2332,38 @@ func (core *Core) v2API() *route.Router {
 			return
 		}
 
-		limit, err := strconv.Atoi(r.Param("limit", "0"))
-		if err != nil || limit < 0 {
+		limit, err := strconv.Atoi(r.Param("limit", "30"))
+		if err != nil || limit < 0 || limit > 30 {
 			r.Fail(route.Bad(err, "Invalid limit parameter given"))
 			return
 		}
 
-		tasks, err := core.DB.GetAllTasks(
-			&db.TaskFilter{
+		// check to see if we're offseting task requests
+		paginationDate, err := strconv.ParseInt(r.Param("before", "0"), 10, 64)
+
+		var filter db.TaskFilter
+		if paginationDate > 0 {
```

Comment:
Do you actually need the if?  if paginationDate is 0, there's no difference in the filter assignment; `BeforeCreatedDate: 0` is the same as not setting the field, since the zero value of an integer is 0.  We can probably remove the conditional split, and save the duplicated code.